uplink/piecestore/upload.go

```diff
@@ -138,7 +145,14 @@ func (client *Upload) Write(data []byte) (written int, err error) {
 			},
 		})
 		if err != nil {
-			err = ErrProtocol.Wrap(err)
+			_, closeErr := client.stream.CloseAndRecv()
```

Comment:
This still has the `CloseAndRecv`, which it should not do. I think the previous code was more correct... i.e. the code that was initially here.

Reply:
This is the change that allowed to make the error visible correctly to the uplink.
I'll manually test again to confirm it.

Reply:
With the original code it reports:

```
2019-07-30T17:11:19.882+0200    DEBUG   ecclient        ecclient/client.go:319  Failed uploading piece to node     {"pieceID": "mhiOqkhU45GSOh0S/aGnkM91i2P5475MOc2HQr1s8WU=", "nodeID": "Fiw9mqj0G3F5ITBlBs/+aUUCrtw9squa9scnSVGKjQA=", "nodeAddress": "127.0.0.1:14010", "error": "protocol: EOF", "errorVerbose": "protocol: EOF\n\tstorj.io/storj/uplink/piecestore.(*Upload).Write:148\n\tbufio.(*Writer).Flush:590\n\tbufio.(*Writer).Write:626\n\tstorj.io/storj/uplink/piecestore.(*BufferedUpload).Write:32\n\tstorj.io/storj/uplink/piecestore.(*LockingUpload).Write:84\n\tio.copyBuffer:404\n\tio.Copy:364\n\tstorj.io/storj/internal/sync2.Copy:22\n\tstorj.io/storj/uplink/ecclient.(*ecClient).putPiece:303\n\tstorj.io/storj/uplink/ecclient.(*ecClient).Put.func1:103"}
```

With theses changes

```
2019-07-30T17:12:25.749+0200    DEBUG   ecclient        ecclient/client.go:319  Failed uploading piece to node  {"pieceID": "/BdlGqdOJ2pkytld+/UfnIZH4WFh2GD9+XmVPU6Tk9U=", "nodeID": "Fiw9mqj0G3F5ITBlBs/+aUUCrtw9squa9scnSVGKjQA=", "nodeAddress": "127.0.0.1:14010", "error": "protocol: rpc error: code = PermissionDenied desc = untrusted: trust:: satellite \"1JJFbFzkuoGh6igpJu3oH9YPKgLB1cBwf9zvpJsngdyg8RQPAK\" is untrusted\n\tstorj.io/storj/storagenode/trust.(*Pool).VerifySatelliteID:66\n\tstorj.io/storj/storagenode/piecestore.(*Endpoint).verifyOrderLimit:57\n\tstorj.io/storj/storagenode/piecestore.(*Endpoint).Upload:221\n\tstorj.io/storj/pkg/pb._Piecestore_Upload_Handler:701\n\tstorj.io/storj/pkg/server.logOnErrorStreamInterceptor:23\n\tgoogle.golang.org/grpc.(*Server).processStreamingRPC:1209\n\tgoogle.golang.org/grpc.(*Server).handleStream:1282\n\tgoogle.golang.org/grpc.(*Server).serveStreams.func1.1:717", "errorVerbose": "protocol: rpc error: code = PermissionDenied desc = untrusted: trust:: satellite \"1JJFbFzkuoGh6igpJu3oH9YPKgLB1cBwf9zvpJsngdyg8RQPAK\" is untrusted\n\tstorj.io/storj/storagenode/trust.(*Pool).VerifySatelliteID:66\n\tstorj.io/storj/storagenode/piecestore.(*Endpoint).verifyOrderLimit:57\n\tstorj.io/storj/storagenode/piecestore.(*Endpoint).Upload:221\n\tstorj.io/storj/pkg/pb._Piecestore_Upload_Handler:701\n\tstorj.io/storj/pkg/server.logOnErrorStreamInterceptor:23\n\tgoogle.golang.org/grpc.(*Server).processStreamingRPC:1209\n\tgoogle.golang.org/grpc.(*Server).handleStream:1282\n\tgoogle.golang.org/grpc.(*Server).serveStreams.func1.1:717\n\tstorj.io/storj/uplink/piecestore.(*Upload).Write:153\n\tbufio.(*Writer).Flush:590\n\tbufio.(*Writer).Write:626\n\tstorj.io/storj/uplink/piecestore.(*BufferedUpload).Write:32\n\tstorj.io/storj/uplink/piecestore.(*LockingUpload).Write:84\n\tio.copyBuffer:404\n\tio.Copy:364\n\tstorj.io/storj/internal/sync2.Copy:22\n\tstorj.io/storj/uplink/ecclient.(*ecClient).putPiece:303\n\tstorj.io/storj/uplink/ecclient.(*ecClient).Put.func1:103"}

````

Which is mostly what we wanted to solve with this PR.

Reply:
Ok, let's go with your version for now.